experimental/zerotrie/src/builder/nonconst/builder.rs

```diff
@@ -239,7 +267,17 @@ impl<S: TrieBuilderStore> ZeroTrieBuilder<S> {
             if ascii_i == key_ascii && ascii_j == key_ascii {
                 let len = self.prepend_ascii(key_ascii)?;
                 current_len += len;
-                debug_assert!(i == new_i || i == new_i + 1);
+                if matches!(self.options.mixed_case_mode, MixedCaseMode::Reject) {
+                    if i == new_i + 2 {
+                        // This can happen if two strings were picked up, each with a different case
```

Comment:
question: how does the condition imply this?

(I'm not clear on how the i/new_i relationship works here)

Reply:
The behavior of i/j is documented in experimental/zerotrie/src/builder/mod.rs

Suppose you have strings like

```
ab
abc
abd
```

When prefix_len is 3, i and j get set to bracket "abc" and "abd". When we then set prefix_len to 2, we pick up "ab" since it has the same prefix. Note that all strings have the same prefix "ab". The only string that can increase the bracket size is indeed "ab", because if it were longer it would have already been in the bracket, if it were shorter it wouldn't be picked up with the prefix length, and if it were different it would be on a different prefix. Now, with mixed case, we could have

```
ab
aB
abc
abd
```

In this case, there are 2 strings we pick up. The only way this happens is if there are identical strings with different casing.

Reply:
aha. thanks!